cpp/src/arrow/filesystem/s3fs_benchmark.cc

```diff
@@ -141,16 +142,31 @@ class MinioFixture : public benchmark::Fixture {
   }
 
   /// Make an object with Parquet data.
+  /// Appends integer columns to the beginning (to act as indices).
   Status MakeParquetObject(const std::string& path, int num_columns, int num_rows) {
-    std::vector<std::shared_ptr<ChunkedArray>> columns(num_columns);
-    std::vector<std::shared_ptr<Field>> fields(num_columns);
-    for (int i = 0; i < num_columns; ++i) {
+    std::vector<std::shared_ptr<ChunkedArray>> columns;
+    std::vector<std::shared_ptr<Field>> fields;
+
+    {
+      arrow::random::RandomArrayGenerator generator(0);
+      std::shared_ptr<Array> values = generator.Int64(num_rows, 0, 1e10, 0);
+      columns.push_back(std::make_shared<ChunkedArray>(values));
+      fields.push_back(::arrow::field("timestamp", values->type()));
+    }
+    {
+      arrow::random::RandomArrayGenerator generator(1);
+      std::shared_ptr<Array> values = generator.Int32(num_rows, 0, 1e9, 0);
+      columns.push_back(std::make_shared<ChunkedArray>(values));
+      fields.push_back(::arrow::field("val", values->type()));
+    }
+
+    for (int i = 0; i < num_columns; i++) {
       arrow::random::RandomArrayGenerator generator(i);
       std::shared_ptr<Array> values = generator.Float64(num_rows, -1.e10, 1e10, 0);
       std::stringstream ss;
       ss << "col" << i;
-      columns[i] = std::make_shared<ChunkedArray>(values);
-      fields[i] = ::arrow::field(ss.str(), values->type());
+      columns.push_back(std::make_shared<ChunkedArray>(values));
+      fields.push_back(::arrow::field(ss.str(), values->type()));
     }
```

Comment:
Aside: this should stretch of code seems to by crying out for a simpler table builder API.